tamagotchi/src/lib.rs

```diff
@@ -1,23 +1,143 @@
 #![no_std]
 use gstd::{debug, exec, msg, prelude::*};
-use tamagotchi_io::TamagotchiState;
+use store_io::{StoreAction, StoreEvent};
+use tamagotchi_io::{TamagotchiState, TmAction, TmEvent};
 
 static mut STATE: Option<TamagotchiState> = None;
 
 #[no_mangle]
 extern "C" fn init() {
+    let current_block_height = exec::block_height() as u64;
     let tamagotchi = TamagotchiState {
         name: msg::load().expect("no name given"),
         date_of_birth: exec::block_timestamp(),
+        owner: msg::source(),
+        fed: tamagotchi_io::MIN_MOOD_VALUE,
+        fed_block: current_block_height,
+        entertained: tamagotchi_io::MIN_MOOD_VALUE,
+        entertained_block: current_block_height,
+        rested: tamagotchi_io::MIN_MOOD_VALUE,
+        rested_block: current_block_height,
+        allowed_account: None,
+        ft_contract: None,
+        transaction_id: 0,
+        ..Default::default()
```

Comment:
No need to set default values.
```suggestion
        ..Default::default()
```